langchain/callbacks/promptlayer_callback.py

```diff
@@ -0,0 +1,125 @@
+import datetime
+from typing import Any, Dict, List, Optional
+
+from promptlayer.utils import get_api_key, promptlayer_api_request
+
+from langchain.callbacks.base import BaseCallbackHandler
+from langchain.chat_models import ChatOpenAI
+from langchain.llms import OpenAI, Replicate
+from uuid import UUID
+from langchain.schema import (
+    AIMessage,
+    BaseMessage,
+    HumanMessage,
+    LLMResult,
+    SystemMessage,
+)
+
+
+class PromptLayerHandler(BaseCallbackHandler):
+    def _convert_message_to_dict(self, message):
+        if isinstance(message, HumanMessage):
+            message_dict = {"role": "user", "content": message.content}
+        elif isinstance(message, AIMessage):
+            message_dict = {"role": "assistant", "content": message.content}
+        elif isinstance(message, SystemMessage):
+            message_dict = {"role": "system", "content": message.content}
+        else:
+            raise ValueError(f"Got unknown type {message}")
+        if "name" in message.additional_kwargs:
+            message_dict["name"] = message.additional_kwargs["name"]
+        return message_dict
+
+    def _create_message_dicts(self, messages):
+        params: Dict[str, Any] = {}
+        message_dicts = [self._convert_message_to_dict(m) for m in messages[0]]
+        return message_dicts, params
+
+    def on_chat_model_start(
+        self,
+        serialized: Dict[str, Any],
+        messages: List[List[BaseMessage]],
+        *,
+        run_id: UUID,
+        parent_run_id: Optional[UUID] = None,
+        tags: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> Any:
+        self.request_start_time = datetime.datetime.now().timestamp()
+
+        self.messages = messages
```

Comment:
Assigning general state like this could lead to undesired race conditions if e.g., a second llm call starts before the first returns